### PR TITLE
Retry queries in grpc on PreparedQueryNotFoundException

### LINKED ISSUE #1065: gRPC's prepared query cache gets out of sync with persistence's query cache
The prepared query cache in gRPC does not currently have a cache invalidation mechanism and because of this when a prepared query is removed from the cache in the persistence module it does not also get removed from the gRPC cache. 

I was able to reproduce this issue by

1. Create a keyspace and table
1. Execute an insert query on the table
1. Drop the keyspace
1. Recreate the same keyspace and table
1. Try to run the same insert query again and it fails

To resolve this we need to either a) remove the local cache in gRPC or b) add cache invalidation in order to keep the caches in sync.

### PR DESCRIPTION
**Pull request by dougwettlaufer**
Resolves #1065

### CODE REVIEW

File: grpc/src/main/java/io/stargate/grpc/service/Service.java [hunk 1]
```diff
@@ -426,7 +428,22 @@ private void executePrepared(
           .whenComplete(
               (result, t) -> {
                 if (t != null) {
-                  handleException(t, responseObserver);
+                  if (t instanceof PreparedQueryNotFoundException) {
```
Comment (ivansenic) [inline grpc/src/main/java/io/stargate/grpc/service/Service.java:431]: Are we handling those in other modules?

Comment (dougwettlaufer) [reply grpc/src/main/java/io/stargate/grpc/service/Service.java:431]: Logic borrowed from persistence-api 
https://github.com/stargate/stargate/blob/6452957cd078f17f1c0067c943e3fc2129f61ceb/persistence-api/src/main/java/io/stargate/db/datastore/PersistenceBackedDataStore.java#L115

File: grpc/src/main/java/io/stargate/grpc/service/Service.java [hunk 2]
```diff
@@ -426,7 +428,22 @@ private void executePrepared(
...
+                    preparedCache.invalidate(prepareInfo);
+
+                    prepareQuery(connection, prepareInfo, parameters.getTracing())
+                        .whenComplete(
+                            (p, t1) -> {
+                              if (t1 != null) {
+                                handleException(t1, responseObserver);
+                              } else {
+                                executePrepared(
```
Comment (dimas-b) [inline grpc/src/main/java/io/stargate/grpc/service/Service.java:440]: Should we be checking for infinite loops here? It's probably not likely, but still... :thinking:

**Comment by mpenick** [comment]
I think we need to also handle unprepared queries in batches.

File: grpc/src/main/java/io/stargate/grpc/service/Service.java (hunk 2 above)
Comment (dougwettlaufer) [reply grpc/src/main/java/io/stargate/grpc/service/Service.java:440]: I thought about that but noticed `PersistenceBackedDataStore.java` doesn't check so I figured we'd be fine without it.

Comment (dimas-b) [reply grpc/src/main/java/io/stargate/grpc/service/Service.java:440]: I do believe the same problem exists there too, but I think it is not likely to result in real issues. The schema would have to change very often or the statement preparation has to have bugs for this infinite loop to manifest. +0

Comment (tomekl007) [reply grpc/src/main/java/io/stargate/grpc/service/Service.java:440]: Maybe we should do an exponential backoff before retry?

**Review by dimas-b — APPROVED** [review]

**Pull request merged**